src/Features/Core/Portable/GenerateEqualsAndGetHashCodeFromMembers/GenerateEqualsAndGetHashCodeAction.cs

```diff
@@ -132,30 +136,16 @@ private async Task<Document> UpdateDocumentAndAddImportsAsync(SyntaxNode oldType
                 return newDocument;
             }
 
-            private async Task<(SyntaxNode oldType, SyntaxNode newType)> AddMethodsAsync(
-                IList<IMethodSymbol> methods,
-                CancellationToken cancellationToken)
-            {
-                var workspace = _document.Project.Solution.Workspace;
-                var syntaxTree = await _document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
-
-                var declarationService = _document.GetLanguageService<ISymbolDeclarationService>();
-                var typeDeclaration = declarationService.GetDeclarations(_containingType)
-                                                        .Select(r => r.GetSyntax(cancellationToken))
-                                                        .First(s => s.FullSpan.IntersectsWith(_textSpan.Start));
```

Comment:
this code was busted in two ways.  First, it was only comparing spans of nodes, not considering that the different declarations could be in different files (hence why we were breaking with partial types).  Second, we don't even need to do any of this as we already have the starting node the user invoked the feature on.  So we can just use that instead.